refactor(napi/parser): remove unnecessary `#[estree(field_order)]` attr

overlookmotel opened the pull request:
`span` is now automatically moved to be last field, so no need to specify field order explicitly here.

Boshen reviewed (commented):
These could just be one PR ...

overlookmotel committed:
refactor(napi/parser): remove unnecessary `#[estree(field_order)]` attr (#14144)

`span` is now automatically moved to be last field, so no need to specify field order explicitly here.
